types/yup/index.d.ts

```diff
@@ -435,3 +514,18 @@ export interface LocaleObject {
     array?: { [key in keyof ArraySchema<any>]?: string };
     object?: { [key in keyof ObjectSchema<any>]?: string };
 }
+
+export type InferType<T> = T extends Schema<infer P>
+    ? InnerInferType<P>
+    : never;
+
+// Shut off automatic exporting after this statement
+export {};
+
+type KeyOfUndefined<T> = {
```

Comment:
I think having these types are clutch!

Reply:
Thanks :-)

When this PR is merged I will do a PR on the yup docs to show how to use this.